src/Receiver.cc

```diff
@@ -94,8 +96,9 @@ Receiver::handleDataPacket(Driver::Packet* packet, IpAddress sourceIp)
     Protocol::MessageId id = header->common.messageId;
 
     MessageBucket* bucket = messageBuckets.getBucket(id);
-    SpinLock::Lock lock_bucket(bucket->mutex);
-    Message* message = bucket->findMessage(id, lock_bucket);
+    Tub<SpinLock::Lock> lock_bucket;
```

Comment:
Unfortunately, it's not so simple. I can't get a reference to `SpinLock::Lock` from a `UniqueLock` so that I can pass it to `findMessage`. However, this won't be a problem once the other pending PR is merged.